src/main/java/org/swisspush/mirror/MirrorRequestHandler.java

```diff
@@ -0,0 +1,255 @@
+                        } else {
+                            sendResponse(HttpResponseStatus.OK.code(), null);
+                        }
+                    } else {
+                        LOG.error("ZipEntryPutter failed while working on ZIP from {}", mirrorPath, done.cause());
+                        sendResponse(HttpResponseStatus.INTERNAL_SERVER_ERROR.code(), done.cause().getMessage());
+                    }
+                });
+                zipEntryPutter.handleNext();
+            });
+        }).exceptionHandler(ex -> {
+            LOG.error("Exception occured in Mirror-Get-Request to {}", mirrorPath, ex);
+            sendResponse(HttpResponseStatus.INTERNAL_SERVER_ERROR.code(), ex.toString());
+        }).end();
+    }
+
+    /**
+     * Replaces the delta parameter
+     * with another valid number.
+     * If the pattern could not be found, the original path is returned instead.
+     *
+     * @param path the original path
+     * @param delta the valid delta value
+     * @return the new path with a valid delta value
+     */
+    protected String replaceInvalidDeltaParameter(String path, long delta) {
+        Matcher matcher = DELTA_PARAMETER_PATTERN.matcher(path);
+        if (matcher.find()) {
+            return matcher.replaceAll("delta=" + delta);
+        }
+
+        return path;
+    }
+
+    private void saveNewDeltaAndThenSendResponse() {
+        JsonObject body = new JsonObject();
+        body.put("x-delta", newDelta);
+        Buffer payload = body.toBuffer();
+
+        LOG.debug("mirror - put x-delta-sync file: {} with value: {}", xDeltaSyncPath, newDelta);
+        selfHttpClient
+                .put(xDeltaSyncPath, putNewDeltaResponse -> {
+                    int s = putNewDeltaResponse.statusCode();
+                    if (s == 200) {
+                        sendResponse(s, null);
+                    } else {
+                        sendResponse(s, putNewDeltaResponse.statusMessage());
+                    }
+                })
+                .putHeader(HttpHeaders.CONTENT_TYPE, HttpHeaderValues.APPLICATION_JSON)
+                .putHeader(HttpHeaders.CONTENT_LENGTH, Integer.toString(payload.length()))
+                .end(payload);
+    }
+
+    private void sendResponse(int responseStatusCode, String reason) {
+        JsonObject body = new JsonObject();
+        body.put("success", HttpResponseStatus.OK.code() == responseStatusCode);
+        if (reason != null) {
+            body.put("reason", reason);
+        }
+        if (zipEntryPutter != null) {
+            body.put("loadedresources", zipEntryPutter.loadedResources);
+        }
+        Buffer payload = body.toBuffer();
+
+        response
+                .setStatusCode(responseStatusCode)
+                .putHeader(HttpHeaders.CONTENT_TYPE, HttpHeaderValues.APPLICATION_JSON)
+                .putHeader(HttpHeaders.CONTENT_LENGTH, Integer.toString(payload.length()))
```

Comment:
Instead of setting the content-length you could set `setChunked(true)` as you already do in an error case. 
Moreover you could set this generally for all responses.